poi-ooxml/src/main/java/org/apache/poi/xwpf/usermodel/XWPFRun.java

```diff
@@ -256,8 +256,23 @@ public void setLang(String lang) {
      */
     @Override
     public boolean isBold() {
+        return isBold(ScriptType.NON_COMPLEX);
```

Comment:
Is there any chance that the the run itself could be marked as having ScriptType.COMPLEX? It feels very messy for the API caller to have to specify this on all the methods like isBold, isItalic, etc.

Reply:
I see. You mean as in an internal state for the script type with a setter and a getter? I think i can do that